web/src/links.jsx

```diff
@@ -5,41 +5,51 @@ import {
   Edit,
   Create,
   SimpleForm,
+  FunctionField,
   TextField,
   EditButton,
   TextInput,
   required,
 } from "react-admin";
 
+const PREFIX = "https://links.zenika.com/link"
+
 export const PostList = (props) => (
   <List {...props}>
     <Datagrid>
-      <TextField source="incoming_link" />
-      <TextField source="outgoing_link" />
+      <FunctionField source="incoming_link" label="Generated link" render={record => addUrlPrefix(record.incoming_link)}/>
+      <TextField source="outgoing_link" label="Destination" />
       <EditButton basePath="/links" />
     </Datagrid>
   </List>
 );
 
 const PostTitle = ({ record }) => {
-  return <span>Link {record ? `"${record.incoming_link}"` : ""}</span>;
+  return <span>Link {record ? `"${addUrlPrefix(record.incoming_link)}"` : ""}</span>;
 };
 
 export const PostEdit = (props) => (
   <Edit title={<PostTitle record={props.record} />} {...props}>
     <SimpleForm>
-      <TextInput disabled source="id" />
-      <TextInput source="incoming_link" validate={[required()]} />
-      <TextInput source="outgoing_link" validate={[required()]} />
+      <TextInput disabled source="id" fullWidth/>
+      <TextInput source="incoming_link" label="Preview"  format={v => addUrlPrefix(v)} disabled fullWidth />
```

Comment:
Let's be more explicit with the parameter name, even if it is short live. `incoming_link` or `incomingLink` seem fine to me. Alternatively you could use `format={addUrlPrefix}` to avoid naming the parameter all together.